### DISCUSSION TITLE
Discord OAuth not working

### DISCUSSION THREAD

**Discussion by JoshTheDeveloperr** [nextauthjs/next-auth#948] [Help]
**Describe the bug**
I added the Discord provider and I double checked the client secret and id but when I click on the Discord button as a sign in option I get this error [link to the error](https://discord.com/oauth2/authorize?response_type=code&prompt=consent&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fapi%2Fauth%2Fcallback%2Fdiscord&scope=identify%20email&state=e34ce3c11a92ed3b43f7f592aee1d79f70698654289a70f11161cdebf4349a91&client_id=786794232115494919)

**Steps to reproduce**
Just create a regular Discord provider.

**Expected behavior**
I should sign in using the Discord user and be redirected to my application authenticated

**Comment by jaaneh** [comment]
This can be caused by you not having set the correct callback URL on the application you created with Discord.

Try following these steps: 
- Go to https://discord.com/developers/applications
- Select the application you are using.
- Select "OAuth 2" from the left sidebar.
- Click "Add Redirect" and add `https://localhost:3000/api/auth/callback/discord`. Make sure domain:port is correct for your Next app.
- Under "OAuth2 URL Generator", select the redirect you just added, and hit Save Changes.
- Restart Next app.

![](https://jaany.xyz/i/ce2yu.png)

**Reply by EnderBombz** [reply]
### Realy, thanks, you saved my life of 3 days of thinking

**Reply by Ahmd2121** [reply]
Bro do u know it not showed in my auhroize apps

**Reply by riolly** [reply]
Also, add the redirect uri with `www` 
For example: https://www.yourdomain.com/api/auth/callback/discord
